### PR TITLE
Added state awareness for changed parameters. #1535

### PR DESCRIPTION
**Pull request by rozyczko**
Both for same category and different categories.

### CODE REVIEW

File: src/sas/qtgui/Perspectives/Fitting/UnitTesting/FittingWidgetTest.py [hunk 1]
```diff
@@ -1525,6 +1525,39 @@ def testReplaceConstraintName(self):
...
+        """
+        # select model: cylinder / cylinder
+        category_index = self.widget.cbCategory.findText("Cylinder")
+        self.widget.cbCategory.setCurrentIndex(category_index)
+
+        model_index = self.widget.cbModel.findText("cylinder")
+        self.widget.cbModel.setCurrentIndex(model_index)
+
+        # modify the initial value of radius (different from default)
+        new_value = "333.0"
```
Comment (krzywon) [inline src/sas/qtgui/Perspectives/Fitting/UnitTesting/FittingWidgetTest.py:1540]: You define the value here but don't use it in later tests.

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py [hunk 2]
```diff
@@ -3740,6 +3748,24 @@ def loadAnalysisFile(self):
         filename = QtWidgets.QFileDialog.getOpenFileName(**kwargs)[0]
         return filename
 
+    def getParameterDict(self):
+        """
+        Gather current fitting parameters as dict
+        """
+        param_list = self.getFitParameters()
+        param_list = self.getFitPage()
```
Comment (krzywon) [inline src/sas/qtgui/Perspectives/Fitting/FittingWidget.py:3756]: Anything returned from self.getFitParameters() is immediately overwritten.

**Comment by gonzalezma** [comment]
Nice functionality. The main limitation is that "memory" is restricted to the current model, so some parameter values can be lost if several models are tried.
Example: Starting with a set of values for the sphere, select polymer_micelle (scale, background, and sld_solvent are passed; sld and radius not, as they do not exist in the polymer_micelle model, so this is OK). If then one comes back to the sphere the initial values for sld and radius are lost and the default values are presented instead.
Note also that the passing of the parameters does not work with models allowing for a variable interface (or number of parameters), such as onion or core_multi_shell. In this case, nothing is passed, not even scale and background and then if one comes back to the sphere model, all the parameters are reset to the default values. 
I don't know how easy/feasible is to maintain a list with the last user defined value for each parameter name and use this instead of just the list of the last model. If it requires much refactoring, this PR could be merged as already an improvement over the present behaviour, and create a new enhancement issue to address this limitation.

**Comment by rozyczko** [comment]
> Nice functionality. The main limitation is that "memory" is restricted to the current model, so some parameter values can be lost if several models are tried.
> Example: Starting with a set of values for the sphere, select polymer_micelle (scale, background, and sld_solvent are passed; sld and radius not, as they do not exist in the polymer_micelle model, so this is OK). If then one comes back to the sphere the initial values for sld and radius are lost and the default values are presented instead.
> Note also that the passing of the parameters does not work with models allowing for a variable interface (or number of parameters), such as onion or core_multi_shell. In this case, nothing is passed, not even scale and background and then if one comes back to the sphere model, all the parameters are reset to the default values.
> I don't know how easy/feasible is to maintain a list with the last user defined value for each parameter name and use this instead of just the list of the last model. If it requires much refactoring, this PR could be merged as already an improvement over the present behaviour, and create a new enhancement issue to address this limitation.

I am working on most of those issues now, will commit to this branch

**Comment by rozyczko** [comment]
To make this work properly (changes preserved on category swap, retain data on polydispersity and various checkboxes) more involving changes need to be made. Let's postpone this to 5.0.4

**Pull request merged**